From acbef0543f72f48ff135864eed181a1111fc0700 Mon Sep 17 00:00:00 2001
From: David Tardon <dtardon@redhat.com>
Date: Wed, 10 Oct 2012 17:42:46 +0200
Subject: [PATCH 4/4] remove all traces of saxon

(cherry picked from commit 725eba738eae4eee6cd9b833d375db5463833119)

Conflicts:
	Makefile.top
	Module_tail_build.mk
	RepositoryExternal.mk
	config_host.mk.in
	configure.ac
	ooo.lst.in
	readlicense_oo/html/LICENSE.html
	saxon/prj/build.lst
	saxon/prj/d.lst
	tail_build/prj/build.lst

Change-Id: I33ca2c27d557d24e3900d7ebf8f3ea2d99b3cfb0
---
 Makefile.top                                     |    1 -
 RepositoryExternal.mk                            |   14 -
 config_host.mk.in                                |    3 -
 configure.in                                     |  138 ---
 filter/prj/build.lst                             |    2 +-
 ooo.lst.in                                       |    5 -
 readlicense_oo/html/THIRDPARTYLICENSEREADME.html |   14 +-
 readlicense_oo/txt/NOTICE                        |    5 -
 readlicense_oo/txt/license.txt                   |   14 -
 saxon/README                                     |    1 -
 saxon/makefile.mk                                |   69 --
 saxon/prj/build.lst                              |    2 -
 saxon/prj/d.lst                                  |    1 -
 saxon/saxon-9.0.0.7-bj.patch                     | 1031 ----------------------
 stax/README                                      |    1 -
 stax/makefile.mk                                 |   78 --
 stax/prj/build.lst                               |    2 -
 stax/prj/d.lst                                   |    1 -
 tail_build/prj/build.lst                         |    2 +-
 19 files changed, 3 insertions(+), 1381 deletions(-)
 delete mode 100644 saxon/README
 delete mode 100644 saxon/makefile.mk
 delete mode 100644 saxon/prj/build.lst
 delete mode 100644 saxon/prj/d.lst
 delete mode 100644 saxon/prj/dmake
 delete mode 100644 saxon/saxon-9.0.0.7-bj.patch
 delete mode 100644 stax/README
 delete mode 100644 stax/makefile.mk
 delete mode 100644 stax/prj/build.lst
 delete mode 100644 stax/prj/d.lst
 delete mode 100644 stax/prj/dmake

diff --git a/Makefile.top b/Makefile.top
index 7451445..c84cc6f 100644
--- a/Makefile.top
+++ b/Makefile.top
@@ -232,7 +232,6 @@ rhino\
 saxon\
 setup_native\
 soltools\
-stax\
 stlport\
 sysui\
 tomcat\
diff --git a/RepositoryExternal.mk b/RepositoryExternal.mk
index 730043c..b8a3cbd 100644
--- a/RepositoryExternal.mk
+++ b/RepositoryExternal.mk
@@ -1698,20 +1698,6 @@ endif
 
 ### Jars ############################################################
 
-ifeq ($(SYSTEM_SAXON),YES)
-
-define gb_JavaClassSet__use_saxon
-$(call gb_JavaClassSet_use_system_jar,$(1),$(SAXON_JAR))
-endef
-
-else # !SYSTEM_SAXON
-
-define gb_JavaClassSet__use_saxon
-$(call gb_JavaClassSet_use_jar,$(1),$(OUTDIR)/bin/saxon9.jar)
-endef
-
-endif # SYSTEM_SAXON
-
 ifeq ($(SYSTEM_BSH),YES)
 
 define gb_JavaClassSet__use_bsh
diff --git a/config_host.mk.in b/config_host.mk.in
index 4fb80d1..f643039 100644
--- a/config_host.mk.in
+++ b/config_host.mk.in
@@ -32,7 +32,6 @@ export BUILD_MOZAB=@BUILD_MOZAB@
 export BUILD_NCPUS=@BUILD_NCPUS@
 export BUILD_PLATFORM=@build@
 export BUILD_POSTGRESQL_SDBC=@BUILD_POSTGRESQL_SDBC@
-export BUILD_STAX=@BUILD_STAX@
 export BUILD_TYPE=@BUILD_TYPE@
 export BUILD_UNOWINREG=@BUILD_UNOWINREG@
 export BUILD_VER_STRING=@BUILD_VER_STRING@
@@ -433,7 +432,6 @@ export RTL_OS=@RTL_OS@
 export RTL_ARCH=@RTL_ARCH@
 export SAC_JAR=@SAC_JAR@
 export SAL_ENABLE_FILE_LOCKING=1
-export SAXON_JAR=@SAXON_JAR@
 export SCPDEFS=@SCPDEFS@
 export SERVLETAPI_JAR=@SERVLETAPI_JAR@
 export SHOWINCLUDES_PREFIX=@SHOWINCLUDES_PREFIX@
@@ -528,7 +526,6 @@ export SYSTEM_POSTGRESQL=@SYSTEM_POSTGRESQL@
 export SYSTEM_PYTHON=@SYSTEM_PYTHON@
 export SYSTEM_REDLAND=@SYSTEM_REDLAND@
 export SYSTEM_RHINO=@SYSTEM_RHINO@
-export SYSTEM_SAXON=@SYSTEM_SAXON@
 export SYSTEM_SERVLETAPI=@SYSTEM_SERVLETAPI@
 export SYSTEM_STDLIBS=@SYSTEM_STDLIBS@
 export SYSTEM_VIGRA=@SYSTEM_VIGRA@
diff --git a/configure.in b/configure.in
index b2ae537..0b286ee 100644
--- a/configure.in
+++ b/configure.in
@@ -1427,16 +1427,6 @@ AC_ARG_WITH(libbase-jar,
         [Specify path to jarfile manually.]),
     LIBBASE_JAR=$withval)
 
-AC_ARG_WITH(system-saxon,
-    AS_HELP_STRING([--with-system-saxon],
-        [Use saxon already on system.]),,
-    [with_system_saxon="$with_system_jars"])
-
-AC_ARG_WITH(saxon-jar,
-    AS_HELP_STRING([--with-saxon-jar=JARFILE],
-        [Specify path to jarfile manually.]),
-    SAXON_JAR=$withval)
-
 AC_ARG_WITH(system-odbc,
     AS_HELP_STRING([--with-system-odbc],
         [Use the odbc headers already on system.]),,
@@ -7066,134 +7056,6 @@ AC_SUBST(SYSTEM_POSTGRESQL)
 AC_SUBST(POSTGRESQL_INC)
 AC_SUBST(POSTGRESQL_LIB)
 
-
-dnl ===================================================================
-dnl Check for system saxon
-dnl ===================================================================
-if test "$with_java" != "no"; then
-AC_MSG_CHECKING([which saxon to use])
-if test "$with_system_saxon" = "yes"; then
-    AC_MSG_RESULT([external])
-    SYSTEM_SAXON=YES
-    if test -z $SAXON_JAR; then
-        AC_CHECK_FILE(/usr/share/java/saxon9.jar,
-            [ SAXON_JAR=/usr/share/java/saxon9.jar ],
-            [
-              AC_CHECK_FILE(/usr/share/java/saxon.jar,
-                [ SAXON_JAR=/usr/share/java/saxon.jar ],
-                [ AC_CHECK_FILE(/usr/share/java/saxon9.jar,
-                    [ SAXON_JAR=/usr/share/java/saxon9.jar ],
-                    [ AC_MSG_ERROR(saxon.jar replacement not found)]
-                  )
-                ]
-              )
-            ]
-          )
-    else
-        AC_CHECK_FILE($SAXON_JAR, [],
-               [AC_MSG_ERROR(saxon.jar replacement not found.)], [])
-    fi
-    if test -n "$SERIALIZER_JAR"; then
-        AC_CHECK_FILE($SERIALIZER_JAR, [],
-               [AC_MSG_ERROR(serializer.jar not found.)], [])
-        AC_SUBST(SERIALIZER_JAR)
-    fi
-
-    dnl Saxon comes in two practically available versions, the out-of-date saxonb which
-    dnl supports the Java extensions that LibreOffice uses, and the up-to-date saxon he
-    dnl "home edition" version, which is crippled to not support those Java extensions.
-    dnl And as an aside the he one also needs to be tweaked to include
-    dnl a META-INF/services to broadcast that it supports the jaxp transform factory
-
-    AC_MSG_CHECKING([if saxon works])
-    cat > saxontest.java <<_ACEOF
-[import javax.xml.transform.TransformerFactory;
-import javax.xml.transform.Transformer;
-import javax.xml.transform.stream.StreamSource;
-import java.io.*;
-
-import net.sf.saxon.FeatureKeys;
-
-class saxontest {
-    public static void main(String[] args) {
-      System.setProperty("javax.xml.transform.TransformerFactory",
-        "net.sf.saxon.TransformerFactoryImpl");
-      try {
-        TransformerFactory tfactory = TransformerFactory.newInstance();
-    // some external saxons (Debian, Ubuntu, ...) have this disabled
-    // per default
-    tfactory.setAttribute(FeatureKeys.ALLOW_EXTERNAL_FUNCTIONS, new Boolean(true));
-        System.out.println("TransformerFactory is" +
-          tfactory.getClass().getName());
-        Transformer transformer = tfactory.newTransformer(
-          new StreamSource(new File(args[0])));
-      } catch(Exception e){
-        e.printStackTrace(System.err);
-        System.exit(-1);
-      }
-      System.exit(0);
-    }
-}
-]
-_ACEOF
-
-    cat > saxontest.xsl<<_ACEOF
-[<?xml version="1.0" encoding="UTF-8"?>
-<xsl:stylesheet version="2.0" xmlns:xsl="http://www.w3.org/1999/XSL/Transform">
- <xsl:template match="/">
-  <xsl:value-of select="math:sqrt(1)" xmlns:math="java:java.lang.Math"/>
- </xsl:template>
-</xsl:stylesheet>
-]
-_ACEOF
-
-    javac_cmd="$JAVACOMPILER -cp $SAXON_JAR saxontest.java 1>&2"
-    AC_TRY_EVAL(javac_cmd)
-    if test $? = 0 -a -f ./saxontest.class; then
-        java_cmd="$JAVAINTERPRETER -cp $SAXON_JAR:. saxontest saxontest.xsl 1>&2"
-        AC_TRY_EVAL(java_cmd)
-        if test $? = 0; then
-            AC_MSG_RESULT([yes])
-        else
-            cat saxontest.java >&5
-            AC_MSG_RESULT([no])
-            AC_MSG_ERROR([Non-functional saxon jar, e.g. crippled saxon-he instead of saxonb])
-        fi
-    else
-        AC_MSG_RESULT([no])
-        cat saxontest.java >&5
-        AC_MSG_ERROR([saxontest could not be compiled, non-functional saxon jar])
-    fi
-else
-    AC_MSG_RESULT([internal])
-    SYSTEM_SAXON=NO
-    NEED_SAXON=TRUE
-fi
-fi
-AC_SUBST(SYSTEM_SAXON)
-AC_SUBST(SAXON_JAR)
-
-if test -n "$NEED_SAXON"; then
-    BUILD_TYPE="$BUILD_TYPE SAXON"
-fi
-
-# ===================================================================
-# Check whether to build stax
-# ===================================================================
-AC_MSG_CHECKING([whether building the stax is required])
-BUILD_STAX=NO
-if test -z "$SOLAR_JAVA"; then
-    AC_MSG_RESULT([no (java disabled)])
-elif test "$SYSTEM_SAXON" = "YES"; then
-    AC_MSG_RESULT([no (already provided by system saxon)])
-elif test -f "./stax/download/jsr173_1.0_api.jar"; then
-    AC_MSG_RESULT([no (will use the prebuilt stax/download/jsr173_1.0_api.jar)])
-else
-    BUILD_STAX=YES
-    AC_MSG_RESULT([yes (no system saxon and no prebuilt copy)])
-fi
-AC_SUBST(BUILD_STAX)
-
 dnl ===================================================================
 dnl Check for system curl
 dnl ===================================================================
diff --git a/filter/prj/build.lst b/filter/prj/build.lst
index 9083dc7..2319f39 100644
--- a/filter/prj/build.lst
+++ b/filter/prj/build.lst
@@ -1,2 +1,2 @@
-fl      filter  :    TRANSLATIONS:translations svtools unotools xmloff cppu tools cppuhelper sal salhelper svx javaunohelper XPDF:xpdf jvmaccess canvas SAXON:saxon LIBXSLT:libxslt basegfx package PYTHON:python NULL
+fl      filter  :    TRANSLATIONS:translations svtools unotools xmloff cppu tools cppuhelper sal salhelper svx javaunohelper XPDF:xpdf jvmaccess canvas LIBXSLT:libxslt basegfx package PYTHON:python NULL
 fl  filter\prj nmake - all fl_prj   NULL
diff --git a/ooo.lst.in b/ooo.lst.in
index 02d14b0..4d5b977 100644
--- a/ooo.lst.in
+++ b/ooo.lst.in
@@ -8,7 +8,6 @@ c63f411b3ad147db2bcce1bf262a0e02-pixman-0.24.4.tar.bz2
 1756c4fa6c616ae15973c104cd8cb256-Adobe-Core35_AFMs-314.tar.gz
 18f577b374d60b3c760a3a3350407632-STLport-4.5.tar.gz
 1f24ab1d39f4a51faf22244c94a6203f-xmlsec1-1.2.14.tar.gz
-24be19595acad0a2cae931af77a0148a-LICENSE_source-9.0.0.7-bj.html
 26b3e95ddf3d9c077c480ea45874b3b8-lp_solve_5.5.tar.gz
 284e768eeda0e2898b0d5bf7e26a016e-raptor-1.4.18.tar.gz
 2a177023f9ea8ec8bd00837605c5df1b-jakarta-tomcat-5.0.30-src.tar.gz
@@ -30,12 +29,9 @@ f101a9e88b783337b20b2e26dfd26d5f-cairo-1.10.2.tar.gz
 7376930b0d3f3d77a685d94c4a3acda8-STLport-4.5-0119.tar.gz
 798b2ffdc8bcfe7bca2cf92b62caf685-rhino1_5R5.zip
 ecb2e37e45c9933e2a963cabe03670ab-curl-7.19.7.tar.gz
-8294d6c42e3553229af9934c5c0ed997-stax-api-1.0-2-sources.jar
 bd30e9cf5523cdfc019b94f5e1d7fd19-cppunit-1.12.1.tar.gz
 a169ab152209200a7bad29a275cb0333-seamonkey-1.1.14.source.tar.gz
-a4d9b30810a434a3ed39fc0003bbd637-LICENSE_stax-api-1.0-2-sources.html
 a7983f859eafb2677d7ff386a023bc40-xsltml_2.1.2.zip
-ada24d37d8d638b3d8a9985e80bc2978-source-9.0.0.7-bj.zip
 af3c3acf618de6108d65fcdc92b492e1-commons-codec-1.3-src.tar.gz
 bc702168a2af16869201dbe91e46ae48-LICENSE_Python-2.6.1
 c441926f3a552ed3e5b274b62e86af16-STLport-4.0.tar.gz
@@ -48,7 +44,6 @@ e61d0364a30146aaa3001296f853b2b9-libxslt-1.1.26.tar.gz
 ea570af93c284aa9e5621cd563f54f4d-bsh-2.0b1-src.tar.gz
 ea91f2fb4212a21d708aced277e6e85a-vigra1.4.0.tar.gz
 dd7dab7a5fea97d2a6a43f511449b7cd-expat-2.1.0.tar.gz
-fb7ba5c2182be4e73748859967455455-README_stax-api-1.0-2-sources.txt
 fca8706f2c4619e2fa3f8f42f8fc1e9d-rasqal-0.9.16.tar.gz
 f872f4ac066433d8ff92f5e316b36ff9-dejavu-fonts-ttf-2.33.zip
 0981bda6548a8c8233ffce2b6e4b2a23-mysql-connector-c++-1.1.0.tar.gz
diff --git a/readlicense_oo/html/THIRDPARTYLICENSEREADME.html b/readlicense_oo/html/THIRDPARTYLICENSEREADME.html
index 46fc32f..4fca83d 100644
--- a/readlicense_oo/html/THIRDPARTYLICENSEREADME.html
+++ b/readlicense_oo/html/THIRDPARTYLICENSEREADME.html
@@ -2110,18 +2110,6 @@ FITNESS FOR A PARTICULAR PURPOSE.</P>
 <P CLASS="western" STYLE="margin-top: 0.3cm; margin-bottom: 0.3cm">This
 file declares SANE application interface. See the SANE standard for a
 detailed explanation of the interface.</P>
-<H2 CLASS="western">Saxon (B)</H2>
-<P CLASS="western" STYLE="margin-top: 0.3cm; margin-bottom: 0.3cm">The
-following software may be included in this product: Saxon (B). Use of
-any of this software is governed by the terms of the license below:</P>
-<P CLASS="western" STYLE="margin-top: 0.3cm; margin-bottom: 0.3cm"><A HREF="#10.Mozilla Public License Version 1.1|outline">Jump
-to MPL Version 1.1</A></P>
-<H2 CLASS="western">Stax</H2>
-<P CLASS="western" STYLE="margin-top: 0.3cm; margin-bottom: 0.3cm">The
-following software may be included in this product: Stax API. Use of
-any of this software is governed by the terms of the license below:</P>
-<P CLASS="western" STYLE="margin-top: 0.3cm; margin-bottom: 0.3cm"><A HREF="#11.Apache License|outline">Jump
-to Apache License Version 2.0</A></P>
 <H2 CLASS="western">STLPort</H2>
 <P CLASS="western" STYLE="margin-top: 0.3cm; margin-bottom: 0.3cm">The
 following software may be included in this product:STLport. Use of
@@ -9429,4 +9417,4 @@ this trademark restriction does not form part of the License.</P>
 <P CLASS="western" STYLE="margin-top: 0.3cm; margin-bottom: 0.3cm">Creative
 Commons may be contacted at <A HREF="http://creativecommons.org/">http://creativecommons.org/</A>.</P>
 </BODY>
-</HTML>
\ No newline at end of file
+</HTML>
diff --git a/readlicense_oo/txt/NOTICE b/readlicense_oo/txt/NOTICE
index 181aea7..8edf0f0 100644
--- a/readlicense_oo/txt/NOTICE
+++ b/readlicense_oo/txt/NOTICE
@@ -139,7 +139,6 @@ As part of the base system this product also includes code from the following
 projects which are licensed under the Apache license:
 - serf
 - redland
-- StAX
 
 The notices from these projects are following:
 
@@ -153,10 +152,6 @@ This product includes Redland software (http://librdf.org/)
 developed at the Institute for Learning and Research Technology,
 University of Bristol, UK (http://www.bristol.ac.uk/).
 
-
-StAX
-[no notices]
-
 ____
 
 This product includes software developed by the OpenSSL Project
diff --git a/readlicense_oo/txt/license.txt b/readlicense_oo/txt/license.txt
index 6acf5b6d..137a892 100644
--- a/readlicense_oo/txt/license.txt
+++ b/readlicense_oo/txt/license.txt
@@ -2057,20 +2057,6 @@ SANE
    This file declares SANE application interface. See the SANE standard
    for a detailed explanation of the interface.
 
-Saxon (B)
-
-   The following software may be included in this product: Saxon (B). Use
-   of any of this software is governed by the terms of the license below:
-
-   See MPL Version 1.1 below
-
-Stax
-
-   The following software may be included in this product: Stax API. Use
-   of any of this software is governed by the terms of the license below:
-
-   See Apache License Version 2.0 below
-
 STLPort
 
    The following software may be included in this product:STLport. Use of
diff --git a/saxon/README b/saxon/README
deleted file mode 100644
index 26344a7..0000000
--- a/saxon/README
+++ /dev/null
@@ -1 +0,0 @@
-XSLT and XQuery Processor from [[http://saxon.sourceforge.net/]]
diff --git a/saxon/makefile.mk b/saxon/makefile.mk
deleted file mode 100644
index 40a7404..0000000
--- a/saxon/makefile.mk
+++ /dev/null
@@ -1,69 +0,0 @@
-#*************************************************************************
-#
-# DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
-# 
-# Copyright 2000, 2010 Oracle and/or its affiliates.
-#
-# OpenOffice.org - a multi-platform office productivity suite
-#
-# This file is part of OpenOffice.org.
-#
-# OpenOffice.org is free software: you can redistribute it and/or modify
-# it under the terms of the GNU Lesser General Public License version 3
-# only, as published by the Free Software Foundation.
-#
-# OpenOffice.org is distributed in the hope that it will be useful,
-# but WITHOUT ANY WARRANTY; without even the implied warranty of
-# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
-# GNU Lesser General Public License version 3 for more details
-# (a copy is included in the LICENSE file that accompanied this code).
-#
-# You should have received a copy of the GNU Lesser General Public License
-# version 3 along with OpenOffice.org.  If not, see
-# <http://www.openoffice.org/license.html>
-# for a copy of the LGPLv3 License.
-#
-#*************************************************************************
-
-PRJ=.
-
-PRJNAME=saxon
-TARGET=saxon
-
-# --- Settings -----------------------------------------------------
-
-.INCLUDE :	settings.mk
-.INCLUDE : antsettings.mk
-
-.IF "$(SOLAR_JAVA)" != ""
-.IF "$(SYSTEM_SAXON)" == "YES"
-all:
-        @echo "An already available installation of saxon should exist on your system."
-    @echo "Therefore the version provided here does not need to be built in addition."
-.ENDIF
-
-# --- Files --------------------------------------------------------
-
-TARFILE_NAME=source-9.0.0.7-bj
-TARFILE_MD5=ada24d37d8d638b3d8a9985e80bc2978
-ADDITIONAL_FILES=build.xml
-TARFILE_ROOTDIR=src
-
-PATCH_FILES=saxon-9.0.0.7-bj.patch
-
-BUILD_ACTION="$(ANT)" $(ANT_FLAGS) -Dsolarbindir=$(SOLARBINDIR) jar-bj
-
-OUT2CLASS= saxon-build$/9.0.0.7$/bj$/saxon9.jar
-
-.ELSE			# $(SOLAR_JAVA)!= ""
-nojava:
-    @echo "Not building $(PRJNAME) because Java is disabled"
-.ENDIF			# $(SOLAR_JAVA)!= ""
-# --- Targets ------------------------------------------------------
-
-.INCLUDE : set_ext.mk
-.INCLUDE : target.mk
-.IF "$(SOLAR_JAVA)" != ""
-.INCLUDE : tg_ext.mk
-.ENDIF
-
diff --git a/saxon/prj/build.lst b/saxon/prj/build.lst
deleted file mode 100644
index 813097c..0000000
--- a/saxon/prj/build.lst
+++ /dev/null
@@ -1,2 +0,0 @@
-xx	saxon	:	solenv stax NULL
-xx	saxon	nmake	-	all	xx_saxon NULL
diff --git a/saxon/prj/d.lst b/saxon/prj/d.lst
deleted file mode 100644
index 0793dad..0000000
--- a/saxon/prj/d.lst
+++ /dev/null
@@ -1 +0,0 @@
-..\%__SRC%\class\saxon9.jar %_DEST%\bin\saxon9.jar
diff --git a/saxon/prj/dmake b/saxon/prj/dmake
deleted file mode 100644
index e69de29..0000000
diff --git a/saxon/saxon-9.0.0.7-bj.patch b/saxon/saxon-9.0.0.7-bj.patch
deleted file mode 100644
index 11d3091..0000000
--- a/saxon/saxon-9.0.0.7-bj.patch
+++ /dev/null
@@ -1,1031 +0,0 @@
---- misc/src/build.xml	2008-07-11 12:11:50.079281000 +0200
-+++ misc/build/src/build.xml	2008-07-11 12:17:29.848740000 +0200
-@@ -1 +1,1027 @@
--dummy
-+<?xml version="1.0" encoding="UTF-8"?>
-+
-+<project basedir="." default="all" name="Saxon-B The XSLT and XQuery Processor"
-+         xmlns:dn="antlib:org.apache.ant.dotnet">
-+         
-+         
-+    <!-- TODO *really* Must publish NOTICES directory with patch releases. Should also
-+          publish doc directory, and add a notice concerning Saxon/MPL itself, as per section 3.6 of MPL 1.0 -->     
-+    
-+    <!--TODO: 1: ensure saxon9api.netmodule is included in the build
-+           2: remove the archive target before shipping this file-->
-+    
-+    <!-- note : Ant 1.7 is needed for this script -->
-+    
-+    <!-- Ant 1.7 introduces a <service> element to allow META-INF/services
-+         directories to be created in JAR files. Unfortunately it is broken,
-+         see http://issues.apache.org/bugzilla/show_bug.cgi?id=41201. The relevant
-+         code here is retained in the form of comments, and can be reinstated
-+         when the bug is fixed. Meanwhile the directories are created manually -->
-+    
-+    <!-- Build file for Saxon. This builds all four for the released products:
-+         Saxon-B and Saxon-SA, on both the Java and .NET platforms. It also builds
-+         the released resources file. A full build requires access to Saxonica
-+         proprietary source code. The open source Saxon-B product can be built for
-+         Java using target "bj"; but building Saxon-B for .NET may require some
-+         disentangling of tasks that depend on Saxon-SA being present. -->
-+         
-+    <!-- Useful targets:
-+    
-+         clean        clears temporary directories
-+         bj           builds executables for Saxon-B 
-+         saj          builds executables for Saxon-B and Saxon-SA 
-+         n            builds executables for Saxon on .NET
-+         build        builds all executables
-+         resources    builds samples and documentation
-+         release      builds ZIP files for published downloads
-+         archive      produces a back-up archive  -->
-+             
-+         
-+    <!-- Naming conventions:
-+         -n   : .NET
-+         -j   : Java
-+         -b   : Saxon-B
-+         -sa  : Saxon-SA
-+         combined as appropriate, so -bj denotes Saxon-B on the Java platform
-+    -->     
-+    
-+    <property name="version" value="9.0.0.7"/>
-+    <property name="versionh" value="9-0-0-7"/>
-+    <property name="release.date" value="2008-07-03"/>
-+    
-+    <!-- following locations are specific to the machine -->
-+
-+		    <!-- root drive for input files-->
-+		    <property name="root" value="."/>
-+		    
-+		    <!-- root drive for generated files-->
-+		    <property name="troot" value="."/>
-+		    
-+		    <!-- directory holding Java source code -->
-+		    <property name="development-j.dir" value="${root}/MyJava/saxon8.x"/>
-+		    
-+		    <!-- directory holding Java source code as in Subversion -->
-+		    <property name="subversion-j.dir" value="${root}"/>		
-+		    
-+		    <!-- directory holding latest patch state of Saxon-SA Java code -->
-+		    <property name="saj-patch.dir" value="${root}/saxon-90-patch"/>				        
-+		    
-+		    <!-- directory holding Java test drivers -->
-+		    <property name="drivers-j.dir" value="${root}/MyJava/test"/>		    
-+		    
-+		    <!-- directories holding C# source code -->
-+		    <property name="vss-n.dir" value="${root}/Documents and Settings/Mike/My Documents/Visual Studio 2005/Projects/Saxon.Api"/>
-+
-+		    <!-- directory holding C# source code as in Subversion -->
-+		    <property name="subversion-n.dir" value="${root}/saxon-svn/latest9.0/bn/csource"/>	
-+		    
-+		    <!-- directory holding XML user documentation -->
-+		    <property name="userdoc.dir" value="${root}/MyJava/doc/saxon8"/>
-+		    
-+		    <!-- directory holding .NET API documentation (which is built manually using NDOC -->
-+		    <property name="dotnetdoc.dir" value="${root}/MyDotNet/dotnetdoc"/>
-+		    
-+		    <!-- directories holding use cases and sample applications -->
-+		    <property name="usercases.dir" value="${root}/MyJava/use-cases"/>
-+		    <property name="samples.dir" value="${root}/MyJava/samples"/>
-+		    
-+		    <!-- directory in which to build the product for release -->
-+		    <property name="product.dir" value="${troot}/saxon-build/${version}"/>
-+		    
-+		    <!-- directory in which to build the issued .ZIP files -->
-+		    <property name="issue.dir" value="${troot}/saxon-issues/saxon-issue-${release.date} (${version})"/>
-+		    
-+		    <!-- location of the C# compiler -->
-+		    <property name="csharp.compiler" value="c:/Windows/Microsoft.NET/Framework/v1.1.4322/csc.exe"/>   
-+		    
-+    
-+    <!-- following locations are for IKVMC code used to build the .NET product -->
-+    
-+    <property name="development-ikvm.dir" value="${root}/MyDotNet"/>
-+    <property name="ikvm.dir" value="${development-ikvm.dir}/ikvm-0.32.0.0"/>
-+    <property name="generated.stubs.dir" value="${development-ikvm.dir}/stubs"/>
-+    <property name="saxondotnet.snk" value="${development-ikvm.dir}/make/saxondotnet.snk"/>
-+    
-+    <!-- Precondition for .NET compilation: IKVM is installed.
-+         * Download from www.ikvm.net
-+         * Unzip into ${development-ikvm.dir}
-+         * Register in the GAC (access the Microsoft .NET Framework Configuration item in the 
-+                                Windows Control Panel, and add the assemblies to the Assembly Cache. 
-+                                At a minimum, install the IKVM.GNU.Classpath.dll and IKVM.Runtime.dll)
-+         * Generate stubs in ${generated.stubs.dir} 
-+           for System.Xml.dll, System.Security.dll, and IKVM.GNU.Classpath.dll
-+    -->       
-+    
-+    <!-- following locations are relative to the build file -->
-+        
-+    <property name="lib.dir" value="libs"/>
-+    <property name="temp.dir" value="temp"/>
-+    <property name="archive.dir" value="archive"/>
-+    <property name="services.dir" value="services"/>
-+    <property name="temp-src-j.dir" value="${temp.dir}/j/source"/>
-+    <property name="temp-src-nj.dir" value="${temp.dir}/n/jsource"/>
-+    <property name="temp-src-nc.dir" value="${temp.dir}/n/csource"/>
-+    <property name="classes-j.dir" value="${temp.dir}/j/classes"/> 
-+    <property name="classes-n.dir" value="${temp.dir}/n/classes"/> 
-+    <property name="jar-n.dir" value="${temp.dir}/n/jar"/> 
-+    <property name="dll-n.dir" value="${temp.dir}/n/dll"/>  
-+
-+		<!-- following locations are relative to the destination directory -->
-+		
-+		<property name="build-bj.dir" value="${product.dir}/bj"/>  
-+		<property name="build-saj.dir" value="${product.dir}/saj"/> 
-+		<property name="build-bn.dir" value="${product.dir}/bn"/>  
-+		<property name="build-san.dir" value="${product.dir}/san"/>
-+		<property name="build-resources.dir" value="${product.dir}/resources"/> 
-+	  <property name="javadoc.dir" value="${build-resources.dir}/doc/javadoc"/>   
-+    
-+
-+    <property name="build.debug" value="true"/>
-+    <property name="build.debuglevel" value="source,lines,vars"/>
-+    <property name="build.deprecation" value="true"/>
-+    <property name="build.optimize" value="true"/>
-+    <property name="build.compiler.source" value="1.4"/>
-+    <property name="build.compiler.target" value="1.4"/>
-+    <property name="build.s9api.compiler.source" value="1.5"/>
-+    <property name="build.s9api.compiler.target" value="1.5"/>    
-+    <property name="build.compress" value="false"/>
-+    
-+    <!-- following properties can be switched to determine whether to build from patch source
-+         or from development source -->
-+         
-+    <!-- patch source -->     
-+      
-+    <property name="source-bj.dir" value="${subversion-j.dir}"/>
-+    <property name="source-saj.dir" value="${saj-patch.dir}"/>
-+    <property name="source-api-n.dir" value="${subversion-n.dir}/api/Saxon.Api"/>
-+    <property name="source-cmd-n.dir" value="${subversion-n.dir}/cmd"/>
-+    <property name="source-samples-n.dir" value="${subversion-n.dir}/samples"/>
-+      
-+    <!-- development source -->
-+		<!--        
-+    <property name="source-bj.dir" value="${development-j.dir}"/>
-+    <property name="source-saj.dir" value="${development-j.dir}"/>
-+    <property name="source-api-n.dir" value="${vss-n.dir}"/>
-+    <property name="source-cmd-n.dir" value="${vss-n.dir}"/>
-+    <property name="source-samples-n.dir" value="${vss-n.dir}"/>        
-+    -->    
-+    
-+    <property name="solar.bin" value="${solarbindir}"/>
-+    <path id="classpath-j">
-+        <fileset dir="${solar.bin}">
-+            <include name="jsr173_1.0_api.jar"/>
-+        </fileset>
-+    </path>
-+        
-+    <path id="classpath-n">
-+        <fileset dir="${lib.dir}">
-+            <include name="*.jar"/>
-+        </fileset>
-+        <fileset dir="${ikvm.dir}">
-+        	 <include name="classpath/mscorlib.jar"/>
-+        	 <include name="classpath/System.jar"/>
-+        	 <include name="bin/IKVM.GNU.Classpath.jar"/>
-+        	 <include name="bin/IKVM.Runtime.jar"/>
-+        </fileset>
-+        <fileset dir="${generated.stubs.dir}">	 
-+        	 <include name="System.Xml.jar"/>
-+        	 <include name="System.Security.jar"/>
-+        </fileset>	 
-+    </path>
-+    
-+    <path id="classpath-samples">
-+        <fileset dir="${lib.dir}">
-+            <include name="*.jar"/>
-+        </fileset>
-+        <fileset dir="${build-saj.dir}">
-+            <include name="*.jar"/>
-+        </fileset>
-+        <!-- Substitute build-bj.dir if there is no requirement to compile the sample applications
-+             that depend on Saxon-SA -->
-+    </path> 
-+    
-+    <target name="clean" description="Clean all files other than source code and downloaded jars.">
-+        <delete dir="${temp.dir}"/>
-+        <delete dir="${product.dir}"/>
-+        <mkdir dir="${product.dir}"/>
-+    </target>
-+    
-+    <target name="download" description="Download third-party jar files.">
-+        <get src="http://mirrors.ibiblio.org/pub/mirrors/maven2/dom4j/dom4j/1.6.1/dom4j-1.6.1.jar"
-+            dest="${lib.dir}/dom4j-1.6.1.jar" verbose="true"/>
-+
-+        <get src="http://www.ibiblio.org/maven/jdom/jars/jdom-1.0.jar"
-+            dest="${lib.dir}/jdom-1.0.jar" verbose="true"/>
-+
-+        <get src="http://www.cafeconleche.org/XOM/xom-1.1.jar" dest="${lib.dir}/xom-1.1.jar"
-+            verbose="true"/>
-+            
-+        <get src="http://dist.codehaus.org/stax/jars/stax-api-1.0.1.jar" dest="${lib.dir}/stax-api-1.0.1.jar"
-+            verbose="true"/> 
-+            
-+            <!-- TODO: add servlet.jar, ant.jar -->   
-+            
-+    </target>               
-+    
-+    
-+    <target name="copysource-development" description="Copy relevant source files from development directory">
-+        <copy todir="${temp-src-j.dir}">
-+          <fileset dir="${source-bj.dir}">
-+            <include name="net/sf/saxon/**/*.java"/>
-+            <exclude name="**/dotnet/**"/>
-+            <!--<exclude name="**/s9api/**"/>		 s9api needs Java 5 -->
-+            <include name="net/sf/saxon/**/package.html"/>
-+          </fileset>
-+ <!--       
-+          <fileset dir="${source-saj.dir}">
-+            <include name="com/saxonica/**/*.java"/>
-+            <exclude name="com/saxonica/validate/DotNetVerifier.java"/>
-+            <include name="com/saxonica/**/package.html"/>
-+          </fileset>    
-+ -->             
-+        </copy>
-+        <!--
-+        <replace file="${temp-src-j.dir}/com/saxonica/validate/SchemaAwareConfiguration.java"
-+                 token="/*DOTNETONLY*/"
-+                 value="//"
-+                 summary="yes"/>
-+                 -->  
-+        <replace file="${temp-src-j.dir}/net/sf/saxon/Configuration.java"
-+                 token="/*DOTNETONLY*/"
-+                 value="//"
-+                 summary="yes"/>                
-+        <copy todir="${temp-src-nj.dir}">
-+          <fileset dir="${source-bj.dir}">
-+            <include name="net/sf/saxon/**/*.java"/>
-+            <exclude name="**/ant/**"/>
-+            <exclude name="**/dom/**"/>
-+            <exclude name="**/dom4j/**"/>
-+            <exclude name="**/java/**"/>
-+            <exclude name="**/javax/**"/>
-+            <exclude name="**/jdom/**"/>
-+            <exclude name="**/pull/PullToStax.java"/>
-+            <exclude name="**/pull/StaxBridge.java"/>
-+            <!-- <exclude name="**/s9api/**"/>		s9api needs Java 5 -->
-+            <exclude name="**/xom/**"/>
-+            <exclude name="**/xpath/**"/>
-+            <exclude name="**/xqj/**"/>
-+          </fileset>
-+          <!--
-+					<fileset dir="${source-saj.dir}">
-+            <include name="com/saxonica/extra/*.java"/>
-+            <include name="com/saxonica/fsa/*.java"/>
-+            <include name="com/saxonica/ptree/*.java"/>
-+            <include name="com/saxonica/schema/*.java"/>
-+            <include name="com/saxonica/sdoc/*.java"/>
-+            <include name="com/saxonica/validate/*.java"/>
-+            <include name="com/saxonica/Validate.java"/>
-+          </fileset>-->          
-+        </copy>
-+        <!--
-+        <replace file="${temp-src-nj.dir}/com/saxonica/validate/SchemaAwareConfiguration.java"
-+                 token="/*JAVAONLY*/"
-+                 value="//"
-+                 summary="yes"/> 
-+                 --> 
-+        <replace file="${temp-src-nj.dir}/net/sf/saxon/Configuration.java"
-+                 token="/*JAVAONLY*/"
-+                 value="//"
-+                 summary="yes"/>
-+        <!-- when building from development source
-+        <copy todir="${temp-src-nc.dir}/api">
-+          <fileset dir="${source-api-n.dir}">
-+            <include name="Saxon.Api/*.cs"/>
-+          </fileset>
-+        </copy>
-+        
-+        <copy todir="${temp-src-nc.dir}/cmd">
-+          <fileset dir="${source-cmd-n.dir}/Transform">
-+            <include name="*.cs"/>
-+          </fileset>
-+          <fileset dir="${source-cmd-n.dir}/Query">
-+            <include name="*.cs"/>
-+          </fileset>
-+          <fileset dir="${source-cmd-n.dir}/Validate">
-+            <include name="*.cs"/>
-+          </fileset>
-+        </copy>
-+        <copy todir="${temp-src-nc.dir}/samples">
-+          <fileset dir="${source-samples-n.dir}/XQueryExamples">
-+            <include name="*.cs"/>
-+          </fileset>
-+          <fileset dir="${source-samples-n.dir}/XPathExample">
-+            <include name="*.cs"/>
-+          </fileset>
-+          <fileset dir="${source-samples-n.dir}/XsltExamples">
-+            <include name="*.cs"/>
-+          </fileset>
-+          <fileset dir="${source-samples-n.dir}/SchemaExamples">
-+            <include name="*.cs"/>
-+          </fileset>
-+          <fileset dir="${source-samples-n.dir}/XsltTestSuiteDriver">
-+            <include name="*.cs"/>
-+          </fileset>
-+          <fileset dir="${source-samples-n.dir}/XQueryTestSuiteDriver">
-+            <include name="*.cs"/>
-+          </fileset> 
-+          <fileset dir="${source-samples-n.dir}/SampleExtensions">
-+            <include name="*.cs"/>
-+          </fileset>       
-+        </copy>
-+        -->
-+        <!-- when building from patch source 
-+        <copy todir="${temp-src-nc.dir}/api/Saxon.Api">
-+          <fileset dir="${source-api-n.dir}">
-+            <include name="*.cs"/>
-+          </fileset>
-+        </copy>
-+        <copy todir="${temp-src-nc.dir}/cmd" >
-+          <fileset dir="${source-cmd-n.dir}">
-+            <include name="*.cs"/>
-+          </fileset>
-+        </copy>
-+        <copy todir="${temp-src-nc.dir}/samples">
-+          <fileset dir="${source-samples-n.dir}">
-+            <include name="*.cs"/>
-+          </fileset>
-+        </copy>    
-+        -->
-+        <!-- -->
-+        
-+    </target>    
-+    
-+    <target name="create-meta-inf" description="Create contents of META-INF/services files">    
-+        
-+        <!-- given the Ant 1.7.0 bug, create the services files here for copying into the JARS -->
-+        <mkdir dir="${services.dir}/saxon9/META-INF/services"/>
-+        <echo file="${services.dir}/saxon9/META-INF/services/javax.xml.transform.TransformerFactory"
-+              message="net.sf.saxon.TransformerFactoryImpl"/>
-+        <mkdir dir="${services.dir}/saxon9-xpath/META-INF/services"/>
-+        <echo file="${services.dir}/saxon9-xpath/META-INF/services/javax.xml.xpath.XPathFactory"
-+>net.sf.saxon.xpath.XPathFactoryImpl
-+http\://java.sun.com/jaxp/xpath/dom:    net.sf.saxon.xpath.XPathFactoryImpl
-+http\://saxon.sf.net/jaxp/xpath/om:     net.sf.saxon.xpath.XPathFactoryImpl
-+http\://www.xom.nu/jaxp/xpath/xom:      net.sf.saxon.xpath.XPathFactoryImpl
-+http\://jdom.org/jaxp/xpath/jdom:       net.sf.saxon.xpath.XPathFactoryImpl
-+http\://www.dom4j.org/jaxp/xpath/dom4j: net.sf.saxon.xpath.XPathFactoryImpl
-+</echo> 
-+        <mkdir dir="${services.dir}/saxon9sa/META-INF/services"/>
-+        <echo file="${services.dir}/saxon9sa/META-INF/services/javax.xml.transform.TransformerFactory"
-+              message="com.saxonica.SchemaAwareTransformerFactory"/>     
-+        <mkdir dir="${services.dir}/saxon9sa-jaxp/META-INF/services"/>
-+        <echo file="${services.dir}/saxon9sa-jaxp/META-INF/services/javax.xml.validation.SchemaFactory"
-+>com.saxonica.jaxp.SchemaFactoryImpl
-+http\://www.w3.org/2001/XMLSchema: com.saxonica.jaxp.SchemaFactoryImpl
-+</echo>         
-+                                      
-+
-+    </target>
-+    
-+    <target name="compile-bj" description="Compile Saxon-B classes for Java." depends="copysource-development">
-+        <mkdir dir="${classes-j.dir}"/>
-+        <javac encoding="ISO-8859-1" debug="${build.debug}" debuglevel="${build.debuglevel}" 
-+            deprecation="${build.deprecation}" destdir="${classes-j.dir}"
-+            optimize="${build.optimize}" srcdir="${temp-src-j.dir}" source="${build.compiler.source}"
-+            target="${build.compiler.target}">
-+
-+            <include name="net/**"/>
-+            <exclude name="**/dom/**"/>
-+            <exclude name="**/dom4j/**"/>
-+            <exclude name="**/javax/**"/>
-+            <exclude name="**/jdom/**"/>
-+            <exclude name="**/sql/**"/>
-+            <exclude name="**/xom/**"/>
-+            <exclude name="**/xpath/**"/>
-+            <exclude name="**/xqj/**"/>
-+            <exclude name="**/dotnet/**"/>
-+            <exclude name="**/s9api/**"/>            
-+            
-+            <exclude name="**/ant/**"/>
-+            <exclude name="**/dotnet/**"/>
-+            <exclude name="**/s9api/**"/>
-+            <classpath>
-+                <path refid="classpath-j"/>
-+            </classpath>
-+        </javac>
-+    </target>
-+    
-+    <target name="compile-saj" description="Compile Saxon-SA classes for Java." depends="copysource-development">
-+        <mkdir dir="${classes-j.dir}"/>
-+        <javac debug="${build.debug}"  debuglevel="${build.debuglevel}" 
-+            deprecation="${build.deprecation}" destdir="${classes-j.dir}"
-+            optimize="${build.optimize}" srcdir="${temp-src-j.dir}" source="${build.compiler.source}"
-+            target="${build.compiler.target}">
-+
-+            <include name="com/saxonica/**"/>
-+            <include name="net/sf/saxon/ant/**"/>
-+            <exclude name="com/saxonica/validate/DotNetVerifier.java"/>
-+            <classpath>
-+                <path refid="classpath-j"/>
-+            </classpath>
-+        </javac>
-+    </target>
-+    
-+    <target name="compile-s9api" description="Compile s9api classes for Java." 
-+                                 depends="compile-bj, compile-saj, copysource-development">
-+        <!-- Compiling the s9api package requires JDK 1.5 -->
-+        <!-- The bj and saj source must be compiled first to ensure that it is done using 1.4, otherwise
-+             it will be compiled in 1.5 mode because the classes are referenced from the s9api package -->
-+        <javac debug="${build.debug}" debuglevel="${build.debuglevel}" 
-+            deprecation="${build.deprecation}" destdir="${classes-j.dir}"
-+            optimize="${build.optimize}" srcdir="${temp-src-j.dir}" source="${build.s9api.compiler.source}"
-+            target="${build.s9api.compiler.target}">
-+
-+            <include name="**/s9api/**"/>
-+            <classpath>
-+                <path refid="classpath-j"/>
-+            </classpath>
-+        </javac>
-+    </target>
-+        
-+    <target name="compile-bn" description="Compile Saxon-B classes for .NET" depends="copysource-development">
-+        <mkdir dir="${classes-n.dir}"/>
-+        <javac debug="${build.debug}" debuglevel="${build.debuglevel}" 
-+            deprecation="${build.deprecation}" destdir="${classes-n.dir}"
-+            optimize="${build.optimize}" srcdir="${temp-src-nj.dir}" source="${build.compiler.source}"
-+            target="${build.compiler.target}">
-+
-+            <include name="net/**"/>
-+            <exclude name="**/s9api/**"/>
-+            <classpath>
-+                <path refid="classpath-n"/>
-+            </classpath>
-+        </javac>
-+    </target>
-+    
-+    <target name="compile-san" description="Compile Saxon-SA classes for .NET" depends="copysource-development">
-+        <mkdir dir="${classes-n.dir}"/>
-+        <javac debug="${build.debug}"  debuglevel="${build.debuglevel}" 
-+            deprecation="${build.deprecation}" destdir="${classes-n.dir}"
-+            optimize="${build.optimize}" srcdir="${temp-src-nj.dir}" source="${build.compiler.source}"
-+            target="${build.compiler.target}">
-+
-+            <include name="com/saxonica/**"/>
-+            <exclude name="com/saxonica/validate/JavaVerifier.java"/>
-+            <classpath>
-+                <path refid="classpath-n"/>
-+            </classpath>
-+        </javac>
-+    </target>        
-+
-+    <target name="jar-bj" description="Create Saxon-B jar files for Java" depends="create-meta-inf, compile-bj">
-+        <mkdir dir="${product.dir}/bj"/>
-+        <jar basedir="${classes-j.dir}" compress="${build.compress}" jarfile="${product.dir}/bj/saxon9.jar">
-+            <include name="net/**/*.class"/>
-+            <exclude name="**/ant/**/*.class"/>
-+            <exclude name="**/dom/**/*.class"/>
-+            <exclude name="**/dom4j/**/*.class"/>
-+            <exclude name="**/javax/**/*.class"/>
-+            <exclude name="**/jdom/**/*.class"/>
-+            <exclude name="**/sql/**/*.class"/>
-+            <exclude name="**/xom/**/*.class"/>
-+            <exclude name="**/xpath/**/*.class"/>
-+            <exclude name="**/xqj/**/*.class"/>
-+            <exclude name="**/s9api/**/*.class"/>
-+            <include name="META-INF/**"/>
-+            <manifest>
-+                <attribute name="Project-Name" value="Saxon-B"/>
-+                <attribute name="Main-Class" value="net.sf.saxon.Transform"/>
-+            </manifest>
-+            <metainf dir="${services.dir}/saxon9/META-INF"/>
-+            <!--
-+            <service type="javax.xml.transform.TransformerFactory"
-+                provider="net.sf.saxon.TransformerFactoryImpl"/>
-+            -->    
-+        </jar>
-+        <!--
-+        <jar basedir="${classes-j.dir}" compress="${build.compress}" jarfile="${product.dir}/bj/saxon9-dom.jar">
-+            <include name="net/sf/saxon/dom/*.class"/>
-+        </jar>
-+        <jar basedir="${classes-j.dir}" compress="${build.compress}" jarfile="${product.dir}/bj/saxon9-dom4j.jar">
-+            <include name="net/sf/saxon/dom4j/*.class"/>
-+        </jar>
-+        <jar basedir="${classes-j.dir}" compress="${build.compress}" jarfile="${product.dir}/bj/saxon9-jdom.jar">
-+            <include name="net/sf/saxon/jdom/*.class"/>
-+        </jar>
-+        <jar basedir="${classes-j.dir}" compress="${build.compress}" jarfile="${product.dir}/bj/saxon9-s9api.jar">
-+            <include name="net/sf/saxon/s9api/*.class"/>
-+        </jar>         
-+        <jar basedir="${classes-j.dir}" compress="${build.compress}" jarfile="${product.dir}/bj/saxon9-sql.jar">
-+            <include name="net/sf/saxon/sql/*.class"/>
-+        </jar> 
-+        <jar basedir="${classes-j.dir}" compress="${build.compress}" jarfile="${product.dir}/bj/saxon9-xom.jar">
-+            <include name="net/sf/saxon/xom/*.class"/>
-+        </jar> 
-+        <jar basedir="${classes-j.dir}" compress="${build.compress}" jarfile="${product.dir}/bj/saxon9-xpath.jar">
-+            <include name="net/sf/saxon/xpath/*.class"/>
-+            <metainf dir="${services.dir}/saxon9-xpath/META-INF"/>
-+            <!==
-+            <service type="javax.xml.xpath.XPathFactory">
-+                <provider classname="net.sf.saxon.xpath.XPathFactoryImpl"/>
-+                <provider classname="http\://java.sun.com/jaxp/xpath/dom:    net.sf.saxon.xpath.XPathFactoryImpl"/>
-+                <provider classname="http\://saxon.sf.net/jaxp/xpath/om:     net.sf.saxon.xpath.XPathFactoryImpl"/>
-+                <provider classname="http\://www.xom.nu/jaxp/xpath/xom:      net.sf.saxon.xpath.XPathFactoryImpl"/>
-+                <provider classname="http\://jdom.org/jaxp/xpath/jdom:       net.sf.saxon.xpath.XPathFactoryImpl"/>
-+                <provider classname="http\://www.dom4j.org/jaxp/xpath/dom4j: net.sf.saxon.xpath.XPathFactoryImpl"/>
-+            </service> ==>
-+        </jar>  
-+        <jar basedir="${classes-j.dir}" compress="${build.compress}" jarfile="${product.dir}/bj/saxon9-xqj.jar">
-+            <include name="net/sf/saxon/xqj/*.class"/>
-+            <include name="net/sf/saxon/javax/xml/namespace/*.class"/>
-+            <include name="net/sf/saxon/javax/xml/xquery/*.class"/>
-+        </jar>
-+        -->                 
-+    </target>
-+    
-+    <target name="jar-saj" description="Create Saxon-SA jar files for Java" depends="compile-s9api, jar-bj">
-+        <mkdir dir="${product.dir}/saj"/>
-+        <copy todir="${product.dir}/saj">
-+          <fileset dir="${product.dir}/bj">
-+            <include name="*.jar"/>
-+          </fileset>
-+        </copy>          
-+        <jar basedir="${classes-j.dir}" compress="${build.compress}" jarfile="${product.dir}/saj/saxon9sa.jar">
-+            <include name="net/**/*.class"/>
-+            <exclude name="**/dom/**/*.class"/>
-+            <exclude name="**/dom4j/**/*.class"/>
-+            <exclude name="**/javax/**/*.class"/>
-+            <exclude name="**/jdom/**/*.class"/>
-+            <exclude name="**/s9api/**/*.class"/>
-+            <exclude name="**/sql/**/*.class"/>
-+            <exclude name="**/xom/**/*.class"/>
-+            <exclude name="**/xpath/**/*.class"/>
-+            <exclude name="**/xqj/**/*.class"/>
-+            <include name="com/saxonica/**/*.class"/>
-+            <exclude name="com/saxonica/jaxp/*.class"/>
-+            <exclude name="com.saxonica.CompileQuery.class"/>
-+            <manifest>
-+                <attribute name="Project-Name" value="Saxon-SA"/>
-+            </manifest>
-+            <metainf dir="${services.dir}/saxon9sa/META-INF"/>
-+            <!--
-+            <service type="javax.xml.transform.TransformerFactory"
-+                provider="com.saxonica.SchemaAwareTransformerFactory"/>
-+            -->    
-+        </jar>
-+        <jar basedir="${classes-j.dir}" compress="${build.compress}" jarfile="${product.dir}/saj/saxon9sa-jaxp.jar">
-+            <include name="com/saxonica/jaxp/*.class"/>
-+            <include name="net/sf/saxon/dom/TypeInfoImpl.class"/>
-+            <metainf dir="${services.dir}/saxon9sa-jaxp/META-INF"/>
-+            <!--
-+            <service type="javax.xml.validation.SchemaFactory">
-+                <provider classname="com.saxonica.jaxp.SchemaFactoryImpl"/>
-+                <provider classname="http\://www.w3.org/2001/XMLSchema: com.saxonica.jaxp.SchemaFactoryImpl"/>
-+            </service>
-+            -->
-+        </jar>
-+        <jar basedir="${classes-j.dir}" compress="${build.compress}" jarfile="${product.dir}/saj/saxon9sa-qc.jar">
-+            <include name="com/saxonica/codegen/*.class"/>
-+            <include name="com.saxonica.CompileQuery.class"/>
-+        </jar>
-+        <!-- The Saxon-Ant task is delivered with Saxon-B but it won't compile without Saxon-SA present,
-+             unless modified -->
-+        <jar basedir="${classes-j.dir}" compress="${build.compress}" jarfile="${product.dir}/bj/saxon9-ant.jar">
-+            <include name="net/sf/saxon/ant/*.class"/>
-+        </jar>
-+    </target>
-+    
-+    <target name="jar-bn" description="Create Saxon-B jar files for .NET" depends="compile-bn">
-+        <mkdir dir="${jar-n.dir}"/>
-+        <jar basedir="${classes-n.dir}" compress="${build.compress}" jarfile="${jar-n.dir}/saxon9.jar">
-+            <include name="net/**/*.class"/>
-+        </jar>
-+    </target> 
-+    
-+    <target name="jar-san" description="Create Saxon-SA jar files for .NET" depends="compile-san">
-+        <mkdir dir="${jar-n.dir}"/>
-+        <jar basedir="${classes-n.dir}" compress="${build.compress}" jarfile="${jar-n.dir}/saxon9sa.jar">
-+            <include name="com/saxonica/**/*.class"/>
-+        </jar>
-+    </target>
-+    
-+    <target name="dll-n" description="Create DLLs for .NET" depends="jar-bn, jar-san">
-+        <mkdir dir="${dll-n.dir}"/>
-+        <copy todir="${dll-n.dir}">
-+          <fileset dir="${ikvm.dir}/bin">
-+            <include name="IKVM.GNU.Classpath.dll"/>
-+            <include name="IKVM.Runtime.dll"/>
-+          </fileset>
-+        </copy>
-+        <copy file="${saxondotnet.snk}" todir="${basedir}"/> 
-+        <!--
-+            Invoke IKVMC to cross-compile the JAR files into DLLs
-+        -->           
-+        <exec dir="." executable="ikvmc.cmd" >
-+          <arg line="live"/>  <!-- set to 'debug' for testing, 'live' for final build -->
-+          <arg line="${basedir}/${jar-n.dir}"/>
-+          <arg line="${basedir}/${dll-n.dir}"/>
-+          <arg line="${version}"/>
-+        </exec>
-+    </target>
-+    
-+    <target name="csharp-n" description="Compile C# source for .NET commands and API library"
-+        depends="dll-n"> 
-+
-+				<mkdir dir="${dll-n.dir}/samples"/>
-+				<mkdir dir="${dll-n.dir}/tests"/>
-+				
-+				<exec dir="." executable="csharp-compile.cmd" >
-+          <arg line="${basedir}/${temp-src-nc.dir}"/>
-+          <arg line="${basedir}/${dll-n.dir}"/>
-+          <arg line="${version}"/>
-+        </exec>
-+
-+        
-+<!--        <taskdef 
-+            uri="antlib:org.apache.ant.dotnet">
-+            <classpath>
-+              <pathelement location="c:/javalib/ant-dotnet-1.0/ant-dotnet-1.0.jar"/>
-+            </classpath>
-+          </taskdef>
-+    		<dn:csc xmlns:dn="antlib:org.apache.ant.dotnet"
-+    		          executable="${csharp.compiler}"
-+    	     				srcdir="${temp-src-nc.dir}/cmd"
-+    	     				targettype="exe"
-+    	     				win32icon="c:\MyDotNet\icons\gyfu.ico"
-+    	     				destdir="${dll-n.dir}">
-+    	     	<reference file="${dll-n.dir}/IKVM.GNU.Classpath.dll"/>
-+    	     	<reference file="${dll-n.dir}/IKVM.Runtime.dll"/>
-+    	     	<reference file="${dll-n.dir}/saxon9.dll"/>
-+    	     	<reference file="${dll-n.dir}/saxon9sa.dll"/>
-+    	  </dn:csc>   	
-+    		<dn:csc xmlns:dn="antlib:org.apache.ant.dotnet"
-+    		          executable="${csharp.compiler}"
-+    	     				srcdir="${temp-src-nc.dir}/api"
-+    	     				targettype="module"
-+    	     				win32icon="c:\MyDotNet\icons\gyfu.ico"
-+    	     				destdir="${dll-n.dir}">
-+    	     	<reference file="${dll-n.dir}/IKVM.GNU.Classpath.dll"/>
-+    	     	<reference file="${dll-n.dir}/IKVM.Runtime.dll"/>
-+    	     	<reference file="${dll-n.dir}/saxon9.dll"/>
-+    	     	<reference file="${dll-n.dir}/saxon9sa.dll"/> 
-+    	  </dn:csc> 
-+-->    	    	   	     
-+    </target>
-+    
-+    <target name="basic-userdoc" description="Generate Basic Documentation for release with executable">
-+	  <mkdir dir="${product.dir}/bj"/>
-+	  <copy todir="${product.dir}/bj">
-+        <fileset dir="${userdoc.dir}/welcome">
-+          <include name="doc/**"/>
-+          <include name="notices/**"/>
-+        </fileset>
-+      </copy>
-+      <mkdir dir="${product.dir}/saj"/>
-+      <copy todir="${product.dir}/saj">
-+        <fileset dir="${userdoc.dir}/welcome">
-+          <include name="doc/**"/>
-+          <include name="notices/**"/>
-+        </fileset>
-+      </copy>
-+      <mkdir dir="${product.dir}/bn"/>
-+      <copy todir="${product.dir}/bn">
-+        <fileset dir="${userdoc.dir}/welcome">
-+          <include name="doc/**"/>
-+          <include name="notices/**"/>
-+        </fileset>
-+      </copy>
-+      <mkdir dir="${product.dir}/san"/>
-+      <copy todir="${product.dir}/san">
-+        <fileset dir="${userdoc.dir}/welcome">
-+          <include name="doc/**"/>
-+          <include name="notices/**"/>
-+        </fileset>
-+      </copy>       		     
-+    </target>    
-+    
-+    <target name="product-bn" description="Assemble product.directory for Saxon-B on .NET"
-+       										    depends="csharp-n, dll-n, basic-userdoc">
-+      <mkdir dir="${build-bn.dir}/bin"/> 										    
-+      <copy todir="${build-bn.dir}/bin">
-+        <fileset dir="${dll-n.dir}">
-+     	  	<include name="IKVM.GNU.Classpath.dll"/>
-+          <include name="IKVM.Runtime.dll"/>
-+          <include name="saxon9.dll"/>
-+          <include name="saxon9api.dll"/>
-+          <include name="saxon9api.netmodule"/>
-+          <include name="Query.exe"/>
-+          <include name="Transform.exe"/>
-+        </fileset>
-+      </copy>
-+			<echo file="${build-bn.dir}/bin/install-gac.cmd">
-+				set NET="%PROGRAMFILES%\Microsoft.NET\SDK\v1.1\Bin"
-+				%NET%\gacutil /if IKVM.Runtime.dll
-+				%NET%\gacutil /if IKVM.GNU.Classpath.dll
-+				%NET%\gacutil /if saxon9.dll 
-+				%NET%\gacutil /if saxon9api.dll                
-+      </echo> 
-+    </target>  
-+    
-+    <target name="product-san" description="Assemble product.directory for Saxon-SA on .NET"
-+       										    depends="csharp-n, dll-n, basic-userdoc">
-+      <copy todir="${build-san.dir}/bin">
-+        <fileset dir="${dll-n.dir}">
-+     	  	<include name="IKVM.GNU.Classpath.dll"/>
-+          <include name="IKVM.Runtime.dll"/>
-+          <include name="saxon9.dll"/>
-+          <include name="saxon9sa.dll"/>
-+          <include name="saxon9api.dll"/>
-+          <include name="saxon9api.netmodule"/>
-+          <include name="Query.exe"/>
-+          <include name="Transform.exe"/>
-+          <include name="Validate.exe"/>
-+        </fileset> 
-+      </copy> 
-+      <echo file="${build-san.dir}/bin/install-gac.cmd">
-+				set NET="%PROGRAMFILES%\Microsoft.NET\SDK\v1.1\Bin"
-+				%NET%\gacutil /if IKVM.Runtime.dll
-+				%NET%\gacutil /if IKVM.GNU.Classpath.dll
-+				%NET%\gacutil /if saxon9.dll 
-+				%NET%\gacutil /if saxon9sa.dll 
-+				%NET%\gacutil /if saxon9api.dll             
-+      </echo>             
-+    </target>
-+    
-+    <target name="samples" description="Assemble Cross-Platform Samples and Use-Cases">
-+    	<mkdir dir="${build-resources.dir}/samples"/>
-+      <copy todir="${product.dir}/resources/use-cases">
-+        <fileset dir="${usercases.dir}"/>
-+      </copy>
-+      <copy todir="${product.dir}/resources/samples">
-+        <fileset dir="${samples.dir}">
-+          <include name="analysis/timing-profile.xsl"/>
-+          <include name="data/*.xml"/>
-+          <include name="data/*.xsd"/>
-+          <include name="data/*.dtd"/>
-+          <include name="query/*.xq"/>
-+          <include name="styles/*.xsl"/>
-+          <include name="trax/xml/**/*.xml"/>
-+          <include name="trax/xsl/**/*.xsl"/>
-+          <include name="trax/xsl/**/preamble.txt"/>
-+        </fileset>
-+      </copy>
-+    </target>
-+    
-+    <target name="samples-j" description="Copy and Compile Java Samples"
-+    				depends="jar-saj">
-+      <mkdir dir="${product.dir}/resources/samples/java"/>
-+      <mkdir dir="${product.dir}/resources/samples/java/classes"/>
-+      <copy todir="${product.dir}/resources/samples/java">
-+        <fileset dir="${samples.dir}/java">
-+          <include name="ApplyXPathJAXP.java"/>
-+          <include name="ExampleContentHandler.java"/>
-+          <include name="JDOMExample.java"/>
-+          <include name="PullExamples.java"/>
-+          <include name="QueryAPIExamples.java"/>
-+          <include name="QuickValidator.java"/>
-+          <include name="SaxonServlet.java"/>
-+          <include name="SaxonSAServlet.java"/>
-+          <include name="SchemaValidatorExample.java"/>
-+          <include name="SchemaValidatorHandlerExample.java"/>
-+          <include name="TraxExamples.java"/>
-+          <include name="XMarkBenchmark.java"/>
-+          <include name="XPathExample.java"/>
-+          <include name="XPathExampleDOM.java"/>
-+          <include name="XPathExampleSA.java"/>
-+          <include name="XPathExampleServlet.java"/>
-+          <include name="XQJExamples.java"/>
-+        </fileset>
-+      </copy>
-+			<javac debug="true" debuglevel="source, lines" 
-+			      destdir="${product.dir}/resources/samples/java/classes"
-+            srcdir="${product.dir}/resources/samples/java" 
-+            source="${build.compiler.source}"
-+            target="${build.compiler.target}">
-+        <include name="*.java"/>    
-+        <classpath>
-+            <path refid="classpath-samples"/>
-+        </classpath>
-+     </javac>            
-+    </target>
-+    
-+    <target name="testdrivers-j" description="Copy Java Test Drivers">
-+			<mkdir dir="${product.dir}/resources/drivers/java/test"/>
-+			<copy todir="${product.dir}/resources/drivers/java/test">
-+			  <fileset dir="${drivers-j.dir}">
-+			    <include name="XQueryTestSuiteDriver.java"/> 
-+			    <include name="XsltTestSuiteDriver.java"/> 
-+			    <include name="CanonicalXML.java"/> 
-+			    <include name="XQTSModuleURIResolver.java"/> 
-+			    <include name="XQTSCollectionURIResolver.java"/> 
-+			    <include name="SchemaTestSuiteDriver.java"/> 
-+        </fileset>
-+      </copy>
-+    </target>           
-+    
-+    <target name="samples-n" description="Assemble release samples for .NET"
-+                  depends="csharp-n">
-+      <mkdir dir="${build-resources.dir}/samples/cs"/>
-+      <copy todir="${build-resources.dir}/samples/cs"> 
-+        <fileset dir="${temp-src-nc.dir}/samples">
-+          <include name="*.cs"/>
-+        </fileset>
-+      </copy>
-+      <mkdir dir="${build-resources.dir}/samples/cs/bin"/>
-+      <copy todir="${build-resources.dir}/samples/cs/bin"> 
-+        <fileset dir="${dll-n.dir}/samples">
-+          <include name="*.exe"/>
-+          <include name="*.dll"/>
-+          <include name="*.pdb"/>
-+        </fileset>
-+      </copy>
-+      <copy todir="${build-resources.dir}/samples/cs/bin"> 
-+        <fileset dir="${dll-n.dir}/tests">
-+          <include name="*.exe"/>
-+          <include name="*.dll"/>
-+          <include name="*.pdb"/>
-+        </fileset>
-+      </copy>      
-+    </target>               
-+                  
-+    <target name="product-javadoc" description="Generate Javadoc.">
-+        <javadoc bottom="Copyright (C) Michael H. Kay. All rights reserved."
-+            destdir="${javadoc.dir}" doctitle="saxon9 Javadocs"
-+            noindex="true" notree="true" access="public"
-+            maxmemory="512M" source="1.5">
-+            <packageset dir="${temp-src-j.dir}">
-+                <include name="net/**"/>
-+                <include name="com/**"/>
-+                <exclude name="**/dotnet/**"/>
-+                <exclude name="**/DotNetVerifier.java"/>
-+            </packageset>
-+            <classpath>
-+                <path refid="classpath-j"/>
-+            </classpath>
-+            
-+            <link href="http://java.sun.com/j2se/1.5.0/docs/api/"/>
-+            <link href="http://www.xom.nu/apidocs/"/>
-+            
-+        </javadoc>
-+        <copy file="${userdoc.dir}/fixed/javadoc/stylesheet.css"
-+              tofile="${javadoc.dir}/stylesheet.css"
-+              overwrite="true"/>
-+    </target>
-+    
-+    <target name="product-dotnetdoc" description="Copy .NET API documentation">
-+      <mkdir dir="${build-resources.dir}/doc/dotnetdoc"/>
-+      <copy todir="${build-resources.dir}/doc/dotnetdoc">
-+        <fileset dir="${dotnetdoc.dir}"/>
-+      </copy>
-+    </target>  
-+    
-+    
-+    <target name="release-userdoc" description="Generate User Documentation">
-+			<taskdef name="saxon-xslt" 
-+			         classname="net.sf.saxon.ant.AntTransform" 
-+			         classpath="c:\saxon-build\9.0.0.1\saj\saxon9sa.jar;c:\saxon-licenses"/>
-+			<copy todir="${build-resources.dir}/doc">
-+        <fileset dir="${userdoc.dir}/fixed"/>
-+      </copy> 
-+	  <copy todir="${product.dir}/bj">
-+        <fileset dir="${userdoc.dir}/welcome">
-+          <include name="doc/**"/>
-+          <include name="notices/**"/>
-+        </fileset>
-+      </copy>
-+      <copy todir="${product.dir}/saj">
-+        <fileset dir="${userdoc.dir}/welcome">
-+          <include name="doc/**"/>
-+          <include name="notices/**"/>
-+        </fileset>
-+      </copy>
-+      <copy todir="${product.dir}/bn">
-+        <fileset dir="${userdoc.dir}/welcome">
-+          <include name="doc/**"/>
-+          <include name="notices/**"/>
-+        </fileset>
-+      </copy>
-+      <copy todir="${product.dir}/san">
-+        <fileset dir="${userdoc.dir}/welcome">
-+          <include name="doc/**"/>
-+          <include name="notices/**"/>
-+        </fileset>
-+      </copy>       
-+			<saxon-xslt in="${userdoc.dir}/catalog.xml" 
-+			            style="${userdoc.dir}/render-page2.xsl"
-+			            schemaAware="true"
-+			            xmlVersion="1.0"
-+			            schemaValidation="strict"
-+			            out="${product.dir}/resources/doc/dummy.html"><!--validation="strict" -->
-+			  <param name="destination" expression="offline"/>
-+			</saxon-xslt> 
-+			<zip zipfile="${build-resources.dir}/source-userdoc.zip">
-+			  <fileset dir="${userdoc.dir}">
-+			    <include name="**/*.xml"/>
-+			    <include name="**/*.xsl"/>
-+			    <include name="**/*.xsd"/>
-+			    <include name="**/*.css"/>
-+			    <include name="fixed/**"/>
-+			    <include name="welcome/**"/>
-+			  </fileset>
-+			</zip>  			     
-+    </target>
-+    
-+    <target name="release-source" description="Generate Zip file containing issued Source Code">
-+      <mkdir dir="${build-resources.dir}"/>
-+      <zip zipfile="${build-resources.dir}/source.zip">
-+        <fileset dir="${source-bj.dir}">
-+           <include name="net/**/*.java"/>
-+           <include name="net/**/package.html"/>
-+        </fileset>
-+        <fileset dir="${source-bj.dir}">
-+           <include name="net/sf/saxon/dotnet/*.java"/>
-+           <include name="net/sf/saxon/dotnet/package.html"/>
-+        </fileset>        
-+        <fileset dir="${temp-src-nc.dir}">
-+           <include name="api/**/*.cs"/>
-+           <include name="cmd/**/*.cs"/>
-+        </fileset>   
-+      </zip>
-+      <zip zipfile="${build-resources.dir}/sa-source.zip">
-+        <fileset dir="${source-saj.dir}">
-+           <include name="com/saxonica/**/*.java"/>
-+           <include name="com/saxonica/**/package.html"/>
-+        </fileset>
-+      </zip>
-+    </target>
-+    
-+    <target name="resources" description="Generate contents of released resources file"
-+      	depends="release-userdoc, product-javadoc, product-dotnetdoc,
-+      	         samples, samples-j, testdrivers-j, samples-n, release-source">
-+      <mkdir dir="${product.dir}/resources/build"/>
-+      <copy todir="${product.dir}/resources/build">
-+        <fileset dir="${basedir}">
-+          <include name="build.xml"/>
-+          <include name="ikvmc.cmd"/>
-+          <include name="csharp-compile.cmd"/>
-+        </fileset>  
-+      </copy>
-+      <copy todir="${product.dir}/resources">
-+        <fileset dir="${userdoc.dir}/welcome">
-+          <include name="notices/**"/>
-+        </fileset>
-+      </copy>
-+    </target>  
-+      
-+    <target name="bj" depends="compile-bj, jar-bj, basic-userdoc"
-+        description="Generate Saxon-B for the Java platform"/>
-+
-+    <target name="saj" depends="bj, jar-saj, basic-userdoc"
-+        description="Generate Saxon-SA for the Java platform"/>
-+               
-+    <target name="n" depends="product-bn, product-san"
-+    		description="Generate Saxon for the .NET platform"/>    
-+
-+    <target name="build" description="Complete Saxon product build"
-+      	depends="bj, saj, n"/> <!-- add resources -->
-+      
-+    <target name="release" description="Create .ZIP files for release" depends="build"
-+        >  <!--depends="build, resources"-->
-+      <mkdir dir="${issue.dir}"/> 
-+      <zip zipfile="${issue.dir}/saxonb${versionh}j.zip">
-+        <fileset dir="${product.dir}/bj"/>
-+      </zip>
-+      <zip zipfile="${issue.dir}/saxonb${versionh}n.zip">
-+        <fileset dir="${product.dir}/bn"/>
-+      </zip>
-+			<zip zipfile="${issue.dir}/saxonsa${versionh}j.zip">
-+        <fileset dir="${product.dir}/saj"/>
-+      </zip>                     
-+      <zip zipfile="${issue.dir}/saxonsa${versionh}n.zip">
-+        <fileset dir="${product.dir}/san"/>
-+      </zip>
-+      <!--
-+      <zip zipfile="${issue.dir}/saxon-resources${versionh}.zip">
-+        <fileset dir="${product.dir}/resources"/>
-+      </zip>
-+      -->
-+    </target>
-+    
-+    <target name="archive" description="Archive everything needed for backup"> 
-+      <mkdir dir="${archive.dir}"/>
-+      <copy todir="${archive.dir}">
-+       <fileset dir="${issue.dir}"/>
-+       <fileset dir="${development-j.dir}" includes="sa-source.zip"/>
-+       <fileset dir="${build-resources.dir}" includes="source.zip"/>
-+      </copy>
-+      <zip zipfile="${archive.dir}/saxontests.zip">
-+        <fileset dir="e:/tests">
-+          <include name="testsuite/*.*"/>
-+          <include name="java/**/*.java"/>
-+          <include name="xqts-current/SaxonDriver/**"/>
-+          <include name="xqts-extra/**"/>
-+          <include name="xslts_1_0_4/**/SaxonDriver/**"/>
-+        </fileset>  
-+        <fileset dir="e:/tests/java" includes ="**/*.java"/>
-+      </zip>
-+      <zip zipfile="${archive.dir}/licensetool.zip">
-+        <fileset dir="c:/MyJava/licensetool"/>
-+      </zip>
-+      <zip zipfile="${archive.dir}/licenses.zip">
-+        <fileset dir="c:/saxon-licenses"/>
-+      </zip>
-+    </target>  
-+      
-+       
-+    
-+</project>
diff --git a/stax/README b/stax/README
deleted file mode 100644
index 13e69fb..0000000
--- a/stax/README
+++ /dev/null
@@ -1 +0,0 @@
-Library implementing the Streaming API for XML. (CH: Which one, probably the one by Sun???)
diff --git a/stax/makefile.mk b/stax/makefile.mk
deleted file mode 100644
index ff3238b..0000000
--- a/stax/makefile.mk
+++ /dev/null
@@ -1,78 +0,0 @@
-#*************************************************************************
-#
-# DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
-# 
-# Copyright 2000, 2010 Oracle and/or its affiliates.
-#
-# OpenOffice.org - a multi-platform office productivity suite
-#
-# This file is part of OpenOffice.org.
-#
-# OpenOffice.org is free software: you can redistribute it and/or modify
-# it under the terms of the GNU Lesser General Public License version 3
-# only, as published by the Free Software Foundation.
-#
-# OpenOffice.org is distributed in the hope that it will be useful,
-# but WITHOUT ANY WARRANTY; without even the implied warranty of
-# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
-# GNU Lesser General Public License version 3 for more details
-# (a copy is included in the LICENSE file that accompanied this code).
-#
-# You should have received a copy of the GNU Lesser General Public License
-# version 3 along with OpenOffice.org.  If not, see
-# <http://www.openoffice.org/license.html>
-# for a copy of the LGPLv3 License.
-#
-#*************************************************************************
-
-PRJ=.
-
-PRJNAME=stax
-TARGET=stax
-USE_JAVAVER=TRUE
-
-# --- Settings -----------------------------------------------------
-
-.INCLUDE :	settings.mk
-.INCLUDE : antsettings.mk
-
-.IF "$(SOLAR_JAVA)" != ""
-.IF "$(JAVANUMVER:s/.//)" >= "000100060000" || "$(JDK)"=="gcj"
-all:
-    @echo "Your java version already contains StAX"
-.ENDIF			# "$(JAVANUMVER:s/.//)" >= "000100060000" 
-.IF "$(SYSTEM_SAXON)" == "YES"
-all:
-    @echo "An already available installation of saxon should exist on your system."
-    @echo "Therefore the files provided here does not need to be built in addition."
-.ENDIF
-
-.IF "$(BUILD_STAX)" != "YES"
-
-$(OUT)$/class$/jsr173_1.0_api.jar : $(PRJ)$/download$/jsr173_1.0_api.jar
-    +$(COPY) $< $@
-
-.ELSE 			#  "$(BUILD_STAX)" != "YES"
-
-# --- Files --------------------------------------------------------
-TARFILE_NAME=stax-api-1.0-2-sources
-TARFILE_MD5=8294d6c42e3553229af9934c5c0ed997
-TARFILE_ROOTDIR=src
-
-BUILD_ACTION=$(JAVAC) javax$/xml$/stream$/*.java && jar -cf jsr173_1.0_api.jar javax$/xml$/stream$/*.class javax$/xml$/stream$/events$/*.class javax$/xml$/stream$/util$/*.class
-
-OUT2CLASS=jsr173_1.0_api.jar
-.ENDIF			#  "$(BUILD_STAX)" != "YES"
-
-.ELSE			# $(SOLAR_JAVA)!= ""
-nojava:
-    @echo "Not building $(PRJNAME) because Java is disabled"
-.ENDIF			# $(SOLAR_JAVA)!= ""
-# --- Targets ------------------------------------------------------
-
-.INCLUDE : set_ext.mk
-.INCLUDE : target.mk
-.IF "$(SOLAR_JAVA)" != ""
-.INCLUDE : tg_ext.mk
-.ENDIF
-
diff --git a/stax/prj/build.lst b/stax/prj/build.lst
deleted file mode 100644
index cbfc6ea..0000000
--- a/stax/prj/build.lst
+++ /dev/null
@@ -1,2 +0,0 @@
-tx stax : solenv NULL
-tx stax nmake - all tx_stax NULL
diff --git a/stax/prj/d.lst b/stax/prj/d.lst
deleted file mode 100644
index a4fd204..0000000
--- a/stax/prj/d.lst
+++ /dev/null
@@ -1 +0,0 @@
-..\%__SRC%\class\jsr*.jar %_DEST%\bin
diff --git a/stax/prj/dmake b/stax/prj/dmake
deleted file mode 100644
index e69de29..0000000
diff --git a/tail_build/prj/build.lst b/tail_build/prj/build.lst
index c96ae74..0dced2b 100644
--- a/tail_build/prj/build.lst
+++ b/tail_build/prj/build.lst
@@ -1,2 +1,2 @@
-tb      tail_build : AFMS:afms APACHE_COMMONS:apache-commons BSH:beanshell BERKELEYDB:berkeleydb BOOST:boost CAIRO:cairo CPPUNIT:cppunit DESKTOP:codemaker CURL:curl EXPAT:expat FONTCONFIG:fontconfig FREETYPE:freetype GRAPHITE:graphite HSQLDB:hsqldb HUNSPELL:hunspell HYPHEN:hyphen ICU:icu DESKTOP:l10ntools JFREEREPORT:jfreereport JPEG:jpeg LCMS2:lcms2 LIBCDR:libcdr LIBCMIS:libcmis LIBEXTTEXTCAT:libexttextcat LIBXML2:libxml2 LIBXMLSEC:libxmlsec LIBXSLT:libxslt LIBWPG:libwpg LIBWPS:libwps LIBWPD:libwpd LIBVISIO:libvisio LPSOLVE:lpsolve MDDS:mdds MOZ:moz MYTHES:mythes NEON:neon NSS:nss OPENSSL:openssl POSTGRESQL:postgresql PYTHON:python REDLAND:redland SAXON:saxon DESKTOP:setup_native TRANSLATIONS:translations VIGRA:vigra XPDF:xpdf ZLIB:zlib bridges cli_ure comphelper cppu cppuhelper external jurt jvmaccess jvmfwk offapi officecfg DESKTOP:rdbmaker readlicense_oo remotebridges RHINO:rhino ridljar sal salhelper solenv soltools stoc sysui ucbhelper udkapi xmlreader xsltml NULL
+tb      tail_build : AFMS:afms APACHE_COMMONS:apache-commons BSH:beanshell BERKELEYDB:berkeleydb BOOST:boost CAIRO:cairo CPPUNIT:cppunit DESKTOP:codemaker CURL:curl EXPAT:expat FONTCONFIG:fontconfig FREETYPE:freetype GRAPHITE:graphite HSQLDB:hsqldb HUNSPELL:hunspell HYPHEN:hyphen ICU:icu DESKTOP:l10ntools JFREEREPORT:jfreereport JPEG:jpeg LCMS2:lcms2 LIBCDR:libcdr LIBCMIS:libcmis LIBEXTTEXTCAT:libexttextcat LIBXML2:libxml2 LIBXMLSEC:libxmlsec LIBXSLT:libxslt LIBWPG:libwpg LIBWPS:libwps LIBWPD:libwpd LIBVISIO:libvisio LPSOLVE:lpsolve MDDS:mdds MOZ:moz MYTHES:mythes NEON:neon NSS:nss OPENSSL:openssl POSTGRESQL:postgresql PYTHON:python REDLAND:redland DESKTOP:setup_native TRANSLATIONS:translations VIGRA:vigra XPDF:xpdf ZLIB:zlib bridges cli_ure comphelper cppu cppuhelper external jurt jvmaccess jvmfwk offapi officecfg DESKTOP:rdbmaker readlicense_oo remotebridges RHINO:rhino ridljar sal salhelper solenv soltools stoc sysui ucbhelper udkapi xmlreader xsltml NULL
 tb tail_build\prj nmake - all tb_prj   NULL
-- 
1.7.11.7

